### PR TITLE
Refactor proposal feed component

### PR DESCRIPTION
**Pull request by Megha-Dev-19**
All three instances use one feed component

### CODE REVIEW

**Comment by Megha-Dev-19** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@Megha-Dev-19 Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/Megha-Dev-19?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/Megha-Dev-19?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/Megha-Dev-19)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/Megha-Dev-19)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: instances/devhub.near/widget/devhub/entity/proposal/Feed.jsx
Comment (elliotBraem) [inline instances/devhub.near/widget/devhub/entity/proposal/Feed.jsx:18]: Yeah, since this widget is defined in devhub.near and it's configuration is dependent on the instance that is using it, this will need to be passed via props and cannot be handled via a trick with aliases/replacements.

Although I would recommend, rather than passing props.instance here, and then doing switch case for query name, hasura role, query variables, filters, etc.... it may be easier and clearer to handle all unique logic in the calling component and pass it as props. I think state updates may perform better as well, since data query is happening in this widget too.

For example, props could be like:

```
{
  "contract": "${config_account}", // since your defining these props in app that uses it
  "getProposal": (proposalId) => {}, // optionally, instead of passing contract,
  "announcement": () => <>...</>,
  "additionalDetails": ({ proposal, rfp }) => <>...</>, // to replace 185-227
  "query": {
    indexerName: "...", // indexer name associated with app
    whereClause: (state) => {}
    handleData: (data) => {} // to replace lines 415 - 433 handleData(result.body.data).then(...)
    hasuraRole: "..."
  },
  "categories": [...],
}
```

File: instances/devhub.near/widget/config/css.jsx
Comment (elliotBraem) [inline instances/devhub.near/widget/config/css.jsx:4]: Also just noticing that there could be a single "CSS" module defined in devhub.near, then Theme just needs to define the color variables

**Comment by petersalomonsen** [comment]
The events proposal tests fails on the filtering functionality. We have the same functionality in the main devhub feed. Do we also have these featured covered in tests? If so, then we can remove theses failing tests. The Submit proposal tests also fails, but I guess that is just because of the text in the button.

**Comment by petersalomonsen** [comment]
@race-of-sloths score 5

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@petersalomonsen, thank you for scoring this pull request in the Race of Sloths!

**Comment by petersalomonsen** [comment]
Maybe consider making the workflows run the proposal feed test ( proposal.spec.js ) on all 3 instances? ( Like adding also proposal.spec.js here for example: https://github.com/NEAR-DevHub/neardevhub-bos/blob/main/.github/workflows/continuous-integration-workflow-events.yml#L41 )

**Review by petersalomonsen — APPROVED** [review]

**Pull request merged**
